gtk3-macros/src/lib.rs

```diff
@@ -21,38 +21,60 @@ use syn::{parse_macro_input, DeriveInput};
 /// Specify that `MyWidget` is using a composite template and load the
 /// template file the `composite_template.ui` file.
 ///
-/// ```compile_fail
-/// use gtk::prelude::*;
-/// use gtk::CompositeTemplate;
-///
-/// #[derive(Debug, Default, CompositeTemplate)]
-/// #[template(file = "composite_template.ui")]
-/// struct MyWidget {
-///     #[template_child]
-///     pub label: TemplateChild<gtk::Label>,
-/// }
-/// ```
-///
 /// Then, in the `ObjectSubclass` implementation you will need to call
 /// `bind_template` in the `class_init` function, and `init_template` in
 /// `instance_init` function.
 ///
+/// ```no_run
+/// use gtk::prelude::*;
+/// use gtk::glib;
+/// use gtk::subclass::prelude::*;
+///
+/// mod imp {
+///     use gtk::prelude::*;
+///     use gtk::glib;
+///     use gtk::subclass::prelude::*;
```

Comment:
```suggestion
///     use super::*;
```
maybe?

Reply:
Imports don't work this way. :p

You need to use `pub use` in order to be able to use another module's imports. Which is definitely not what we want in here. I originally used `pub use` inside the `imp` module but found it unnecessarily complicated so preferred to duplicated instead.

Reply:
For `super` this actually works. Submodules have full visibility of things in a supermodule

Reply:
It doesn't in this case at least, it's the first thing @bilelmoussaoui tried in fact. :)

Reply:
Strange, that's how we do it everywhere in tests and elsewhere. Must be something strange with rustdoc / doctests.

Reply:
it works fine in gtk4-rs btw https://github.com/gtk-rs/gtk4-rs/blob/master/gtk4-macros/src/lib.rs#L45. What I tried wasn't  work before, because of the missing `# fn main() {}`

Reply:
Oh right, maybe it's working now. Sending another PR then!